tests/Faithlife.OAuth.Tests/OAuthHmacSha256Tests.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Faithlife.Utility;
+using NUnit.Framework;
+
+namespace Faithlife.OAuth.Tests
+{
+	public class OAuthHmacSha256Tests
+	{
+		[TestCase("POST&http%3A%2F%2Fexample.com%2Frequest&a2%3Dr%2520b%26a3%3D2%2520q%26a3%3Da%26b5%3D%253D%25253D%26c%2540%3D%26c2%3D%26oauth_consumer_key%3D9djdj82h48djs9d2%26oauth_nonce%3D7d8f3e4a%26oauth_signature_method%3DHMAC-SHA256%26oauth_timestamp%3D137131201%26oauth_token%3Dkkk9d7dh3k39sjv7",
+			"ypAxjNip++Dm0fTM+gCl8wAo6ufSnseu1WHxL7py3BU=", "http://example.com/request?b5=%3D%253D&a3=a&c%40=&a2=r%20b", "POST", "j49sk3j29djd", "dh893hdasih9",
+			"oauth_consumer_key", "9djdj82h48djs9d2", "oauth_token", "kkk9d7dh3k39sjv7", "oauth_signature_method", "HMAC-SHA256", "oauth_timestamp", "137131201", "oauth_nonce", "7d8f3e4a", "c2", "", "a3", "2 q",
+			Description = "RFC 5849, section 3.4.1.1, amended by errata 2550")]
```

Comment:
RFC 5849 mentions nothing about SHA256.

Reply:
Sorry, that was a copy paste, Ill take a closer look at that description

Reply:
I'm glad there are tests, but did these test cases come from anywhere, or were they copy/paste of the existing tests then change the "expected" value until it passed? 

If so, do we have any confidence that the code returns the _right_ values, or just that it returns the values that it returns?

Reply:
This is a good point.  I dont know how the Sha1 expected values were generated, but all I did for the 256 was see the results of the calls.  Im not too sure how I would go about generating the expected values otherwise though.

Reply:
The expected values for SHA1 were taken from the reference specification (linked to in the `Description` for each test). Hence my question earlier if there is an official standard for this. (Or is it "just the same as HMAC-SHA1, but different hash" ?)

Reply:
Is this some vendor's proprietary extension to OAuth 1.0a?

Reply:
I've found a couple of other libraries online that have added HMAC-SHA256 as an extension; we could copy the values from https://github.com/request/oauth-sign/blob/18a2513da6ba7a2c0cd8179170d7c296c7625137/test.js#L8-L81 and cite them as the source of the test. 

(They're probably just as "made up" as our current expected values, but getting the same results as a library in another language provides a little more confidence in correct implementation of the algorithm.)

That is, delete all the tests in this file that have no provenance, and add C# versions of the oauth-sign/test.js tests.

Reply:
Thanks, Ill take a look